web/src/lib/stores/assets-store.svelte.ts

```diff
@@ -1292,12 +1324,91 @@ export class AssetStore {
     return this.buckets[0]?.getFirstAsset();
   }
 
-  async getPreviousAsset(asset: AssetResponseDto): Promise<AssetResponseDto | undefined> {
-    let bucket = await this.#getBucketInfoForAsset(asset);
+  async getPreviousAsset(
+    idable: { id: string },
+    skipTo: 'asset' | 'day' | 'month' | 'year' = 'asset',
+  ): Promise<AssetResponseDto | undefined> {
+    const bucket = this.#findBucketForAsset(idable.id);
     if (!bucket) {
       return;
     }
+    const asset = bucket.findById(idable.id);
+    if (!asset) {
+      return;
+    }
+    switch (skipTo) {
+      case 'day': {
+        return this.#getPreviousDay(asset, bucket);
+      }
+      case 'month': {
+        return this.#getPreviousMonth(asset, bucket);
+      }
+      case 'year': {
+        return this.#getPreviousYear(asset, bucket);
+      }
+      case 'asset': {
+        return this.#getPreviousAsset(asset, bucket);
+      }
+    }
+  }
+
+  async #getPreviousDay(asset: AssetResponseDto, bucket: AssetBucket) {
+    let nextDay = DateTime.fromISO(asset.localDateTime).toUTC().get('day') + 1;
+    const bucketIndex = this.buckets.indexOf(bucket);
+
+    let nextDaygroup;
+    while (nextDay <= 31) {
+      nextDaygroup = bucket.findDateGroupByDay(nextDay);
+      if (nextDaygroup) {
+        break;
+      }
+      nextDay++;
+    }
+    if (nextDaygroup === undefined) {
+      let previousBucketIndex = bucketIndex - 1;
+      while (previousBucketIndex >= 0) {
+        bucket = this.buckets[previousBucketIndex];
+        if (!bucket) {
+          return;
+        }
+        await this.loadBucket(bucket.bucketDate, { cancelable: false });
+        const previous = bucket.lastDateGroup?.intersetingAssets.at(0)?.asset;
+        if (previous) {
+          return previous;
+        }
+        previousBucketIndex--;
+      }
+    } else {
+      return nextDaygroup.intersetingAssets.at(0)?.asset;
+    }
+  }
+
+  async #getPreviousMonth(asset: AssetResponseDto, bucket: AssetBucket) {
+    const bucketIndex = this.buckets.indexOf(bucket);
+    const previousBucket = this.buckets[bucketIndex - 1];
+    if (previousBucket) {
+      await this.loadBucket(previousBucket.bucketDate, { cancelable: false });
+      return previousBucket.dateGroups[0]?.intersetingAssets[0]?.asset;
+    }
+    return;
+  }
+
+  async #getPreviousYear(asset: AssetResponseDto, bucket: AssetBucket) {
+    const nextYear = DateTime.fromISO(asset.localDateTime).toUTC().get('year') + 1;
+    const bIdx = this.buckets.indexOf(bucket);
+
+    for (let idx = bIdx; idx >= 0; idx--) {
```

Comment:
Still strongly advocate for avoiding abbreviations, so maybe just `index` and `bucketIndex`? Or `i` for the loop iterator.